projects/packages/publicize/src/class-publicize.php

```diff
@@ -201,7 +206,13 @@ public function register_update_publicize_connections_xmlrpc_method( $methods )
 	 */
 	public function get_all_connections() {
 		$this->refresh_connections();
-		$connections = Jetpack_Options::get_option( 'publicize_connections' );
+
+		$connections = get_transient( self::JETPACK_SOCIAL_CONNECTIONS_TRANSIENT );
+
+		if ( $connections === false ) {
+			$connections = array();
+		}
+
```

Comment:
Thinking about this some more, if the transient expires and we can't retrieve the connections because of a network error, then it will look the same as if they have no connections, and we won't inform the user of the error.

Currently we would always show a cached version of the connections from the options, which would then show as broken when the test results API call fails. I'm not sure how much of an issue this is, but it could be worth us implementing something so that we can determine the difference between a site having no connections and there being an error preventing us from retrieving them.

Reply:
Perhaps record an error on local whenever we receive an error? I can create a follow-up task for it.